fe/fe-core/src/main/java/com/starrocks/qe/scheduler/WorkerProvider.java

```diff
@@ -0,0 +1,83 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.qe.scheduler;
+
+import com.starrocks.common.Reference;
+import com.starrocks.system.ComputeNode;
+import com.starrocks.system.SystemInfoService;
+import com.starrocks.thrift.TNetworkAddress;
+
+import java.util.Collection;
+import java.util.List;
+
+public interface WorkerProvider {
```

Comment:
The interfaces in this class aren't clear and generic

Reply:
Any suggestion?
This interface only includes three kinds of methods:
- choose backend or compute node.
- reportBackendNotFoundException
- get used worker information.

Reply:
Updated, use workerId instead of address in scheduling and remove some methods.